### DISCUSSION TITLE
setPositions usage issue

### DISCUSSION THREAD

**Discussion by jimingda** [pattacini/study-air-hockey#6] [Q&A] [answered]
In study-2, pos_arm.size() is 7, but the length of std::vector<int>({0, 1, 2, 3, 4, 5, 6, 7}) seems to be 8, should these two number be equal?
```
        std::vector<double> pos_arm(7);
        for (size_t i = 0; i < pos_arm.size(); i++) {
            pos_arm[i] = interp[pos_torso.size() + i]->operator()(reference->getPos()[0]);
        }
        iposd[1]->setPositions(pos_arm.size(), std::vector<int>({0, 1, 2, 3, 4, 5, 6, 7}).data(),
                              pos_arm.data());
```

**Comment by pattacini** [comment, marked as answer]
Good catch @jimingda 👍🏻 

You spotted a bug that I fixed via https://github.com/pattacini/study-air-hockey/commit/35dbfff2d7636fa80f1f6c2a083ff06142cd2d7d.

Nonetheless, the bug wasn't impacting the code as `setPositions` reads up to the number of values specified by the first parameter, which contained already the right number 7.
